daemon/jobs.go

```diff
@@ -75,6 +75,44 @@ func (r *RPC) StartJobs(
 		}
 	}
 
+	actualIP := JobActualIP(r.dm, r.api)
+
+	go func() {
+		call := func(ctx context.Context, isConnected bool) {
+			err := actualIP(ctx, isConnected)
+			if err != nil {
+				log.Println(internal.ErrorPrefix, "actual ip job error")
+			}
+		}
+
+		stateChan, _ := statePublisher.AddSubscriber()
+		var cancel context.CancelFunc
+
+		for ev := range stateChan {
+			switch ev.(type) {
+			case events.DataConnect:
+			case events.DataDisconnect:
+				if cancel != nil {
+					cancel()
+				}
+
+				var ctx context.Context
+				ctx, cancel = context.WithCancel(context.Background())
+
+				if _, ok := ev.(events.DataConnect); ok {
```

Comment:
So this code has a switch on `events` and processes all cases the same way, but inside the case you are checking event type anyway. Maybe it would be easier to follow with something like:
```
switch ev.(type) {
case events.DataConnect:
  go call(ctx, true)
case events.DataDisconnect:
  call(ctx, false)
}
```
And then handle the cancellation outside of the switch?